src/controllers/cron/update_local_smartapi.js

```diff
@@ -153,24 +153,68 @@ const getOpsFromPredicatesEndpoints = async (specs) => {
+        debug(`ERROR getting API Overrides file because ${error}`);
+        return;
     }
-    debug(`Retrieved ${res.data.total} SmartAPI records`);
-    //clean _score fields
-    const hits = res.data.hits;
-    hits.forEach(function(obj){ delete obj._score });
-    writeFunc(localFilePath, JSON.stringify({ hits: hits }), (err) => {
-        if (err) throw err;
-    });
-    const predicatesInfo = await getOpsFromPredicatesEndpoints(res.data.hits);
-    writeFunc(predicatesFilePath, JSON.stringify(predicatesInfo), (err) => {
-        if (err) throw err;
+    debug(overrides)
+    const res = await axios.get(SMARTAPI_URL, { headers: { 'User-Agent': userAgent } }).catch((err) => {
+        if(process.env.API_OVERRIDE === "true" && overrides.conf.only_overrides === true) {
+            debug(`SmartAPI request failed, creating specs from overrides config.`);
+            let hits = []
+            Object.keys(overrides.apis).map(async (id) => {
+                let override; 
+                try {
+                    const filepath = path.resolve(overrides.apis[id]);
+                    override = yaml.load(await readFile(filepath));
+                } catch (filerror) {
+                    debug(`ERROR getting local file override for API ID ${id} because ${filerror}`);
+                    return;
+                }
+                override._id = id;
+                override._meta = {
+                    date_created: undefined,
+                    last_updated: undefined,
+                    url: overrides.apis[id],
+                    username: undefined,
+                };
+                hits.push(override)
+                fs.writeFile(localFilePath, JSON.stringify({ hits: hits }), (err) => {
+                    if (err) throw err;
+                })
+            });
+            debug('Updated local specs with overrides.');
+        } else {
+            debug(`SmartAPI request failed. Error message: ${err.message}`);
+        }
     });
+    if(await res) {
+        const predicatesFilePath = path.resolve(__dirname, '../../../data/predicates.json');
+        const writeFunc = process.env.SYNC_AND_EXIT === "true" ? fs.writeFileSync : fs.writeFile;
+        if (process.env.API_OVERRIDE === "true") {
+            // if(res)
+            // await getAPIOverrides(res.data).catch(err => {
+            //     debug('test')
+            // });
+            await getAPIOverrides(res.data)
+        }
+        debug(`Retrieved ${res.data.total} SmartAPI records`);
+        //clean _score fields
+        const hits = res.data.hits;
+        hits.forEach(function(obj){ delete obj._score });
+        writeFunc(localFilePath, JSON.stringify({ hits: hits }), (err) => {
+            if (err) throw err;
+        });
+        const predicatesInfo = await getOpsFromPredicatesEndpoints(res.data.hits);
+        writeFunc(predicatesFilePath, JSON.stringify(predicatesInfo), (err) => {
+            if (err) throw err;
+        });
```

Comment:
Predicates have to be generated whether the initial smartAPI request succeeds or not -- these are generated using the hits that are acquired from the request and/or from overrides.